### DISCUSSION TITLE
Scripts And Trainers Conflicting Leading To SP Loading Crash/Endless Loop

### DISCUSSION THREAD

**Discussion by InsomniacTv** [scripthookvdotnet/scripthookvdotnet#1459] [Q&A]
I already made a few post on reddit discussing this issue and already found temporary solutions but felt like i needed to tell you because your the one who makes the scripthook.net builds.

When installing Menyoo/Simple/Native or any other trainer mod it will crash when using scripts even if the folder is empty, but when i uninstall the trainer, the script folder will work or when i remove the scripts folder, basically choosing between script or trainer.

Do you know what's happening? 

I tested each nighty build from 89-100 all have similar issues regarding trainers/scripts, some builds makes scripts work others don't, the only version that works for me is 98 and it seems to be wide spread for others too. 

The only trainer i found that works with the script folder is Rampage Trainer on the nighty.98 build, Some people been saying other versions like 100 or 92 96 extra work, but it depends on that persons individual mod build with conflicting results from them, some say there scripts works with menyoo/native/simple others don't, at this point this is out of my expertise regarding this and i already spent 4 days helping people with lack luster results.

Will the 3.6.0 be updated? Because I'm worried a lot of mod/trainer/script creators are unaware of the issue regarding scripthook.net that could lead to more issues down the line for others. 

Same issue applies to some graphics mods seeing most use the 3.6.0 version.

Note the mods folder works fine and i have already done a fresh reinstalled gta 5 and the directory with the latest ScripthookV patch on the newest game version.

Link to post regarding this.

https://www.reddit.com/r/Gta5Modding/comments/1dsp4eq/script_folder_not_working_with_most_trainer_mods/

**Comment by nomakewan** [comment]
3.6.0 will never be updated. 3.7.0 is the next release currently in development (and is what the Nightly builds are working towards).

When you say "Menyoo" are you referring to the old outdated version or the new Menyoo 2.0?

This could be related to issue #1454

**Comment by InsomniacTv** [comment]
Menyoo 2.0 but not just the same goes for most trainers Rampage Simple Native all crash when using scriptss

**Comment by nomakewan** [comment]
Okay, no worries, was just checking.

The fact that some people say that newer builds worked for them and some don't suggests to me that this is the exact same non-deterministic crash that has plagued SHVDN for nearly a year. It's an issue that the developer is well aware of, but fixing it will require serious work as it is related to the threading model in SHVDN's core.

**Reply by RangerJohnson** [reply]
I found a work around to load with trainer and scripts. Make sure you start gta in windowed mode, if you use steam set the startup to -windowed. Then have another window open like task manager or something. Once you clicked to start, alt+tab into the other window(task manager or whatever). Wait for gta to load, once you see the game has loaded, you should be able to play with scripts and trainer enabled. I use the latest version of menyoo and it works perfect with the scripts. I found this method working 95% of the time, if I don’t have gta starting up in background then it crashes or goes into infinite loading screen.

**Reply by nomakewan** [reply]
The windowed workaround doesn't work for everyone; this is a non-deterministic issue and on some machines it will crash no matter what, even if you have no trainer installed (a friend of mine has a rig like this). On my rig I can indeed shorten the amount of time it takes to load into story mode by alt-tabbing away from GTAV while it's loading then alt-tabbing back in.

But yeah. Welcome to non-deterministic crashes due to race conditions in the SHVDN core.
